src/main/resources/assets/computercraft/lua/rom/programs/http/gist.lua

```diff
@@ -0,0 +1,669 @@
+    if handle == nil then print("Failed.") return 3 end
+    if handle.getResponseCode() ~= 200 then print("Failed.") handle.close() return 3 end
+    local meta = json.decode(handle.readAll())
+    handle.close()
+    if meta == nil or meta.files == nil then print("Failed.") return 3 end
+    local f = {}
+    for k in pairs(meta.files) do table.insert(f, k) end
+    table.sort(f)
+    print("Success.")
+    setTextColor(colors.yellow)
+    write("Description: ")
+    setTextColor(colors.white)
+    print(meta.description)
+    setTextColor(colors.yellow)
+    write("Author: ")
+    setTextColor(colors.white)
+    print(meta.owner.login)
+    setTextColor(colors.yellow)
+    write("Revisions: ")
+    setTextColor(colors.white)
+    print(#meta.history)
+    setTextColor(colors.yellow)
+    print("Files in this Gist:")
+    setTextColor(colors.white)
+    textutils.tabulate(f)
+elseif args[1] == "edit" then
+    if #args < 3 then print(helpstr) return 1 end
+    local data = {files = {}, public = true}
+    local id = args[2]
+    if id:find("/") ~= nil then id = id:match("^([0-9A-Fa-f:]+)/.+$") end
+    if id:find(":") ~= nil then id = id:gsub(":", "/") end
+    local i = 3
+    while args[i] ~= nil and args[i] ~= "--" do
+        if data.files[fs.getName(args[i])] then error("Cannot upload files with duplicate names.") end
+        local file = fs.open(shell.resolve(args[i]), "r")
+        if file == nil then data.files[fs.getName(args[i])] = {} else
+            data.files[fs.getName(args[i])] = {content = file.readAll()}
+            file.close()
+        end
+        i = i + 1
+    end
+    if args[i] == "--" then data.description = table.concat({table.unpack(args, i + 1)}, " ") else
+        write("Connecting to api.github.com... ")
+        local handle = http.get("https://api.github.com/gists/" .. id)
+        if handle == nil then print("Failed.") return 3 end
+        if handle.getResponseCode() ~= 200 then print("Failed.") handle.close() return 3 end
+        local meta = json.decode(handle.readAll())
+        handle.close()
+        if meta == nil or meta.files == nil then print("Failed.") return 3 end
+        data.description = meta.description
+        print("Success.")
+    end
+    -- Get authorization
+    local headers = {["Content-Type"] = "application/json"}
+    requestAuth(headers)
+    local jsonfiles = ""
+    for k, v in pairs(data.files) do jsonfiles = jsonfiles .. (jsonfiles == "" and "" or ",\n") .. (v.content == nil and ("    \"%s\": null"):format(k) or ("    \"%s\": {\n      \"content\": %s\n    }"):format(k, json.encode(v.content))) end
+    local jsondata = ([[{
+  "description": %s,
+  "public": true,
+  "files": {
+%s
+  }
+}]]):format(data.description and '"' .. data.description .. '"' or "null", jsonfiles)
+    write("Connecting to api.github.com... ")
+    local handle
+    if http.patch ~= nil then handle = http.patch("https://api.github.com/gists/" .. id, jsondata, headers)
```

Comment:
This should be the following right:

```lua
http.post {
  url = "https://api.github.com/gists/" .. id,
  body = jsondata,
  headers = headers,
  method = "PATCH"
}
```

Neither `http.patch` nor the extended `http.request` syntax are supported in CC:T. The latter is from CCTweaks, not sure where the former is from.

Reply:
`http.patch` is added in CraftOS-PC. I thought the `http.request` syntax was what CC:T used, but I'll swap that out for the correct syntax.

(I'll probably also need to add the table syntax for http.post into CraftOS-PC... but that's irrelevant here.)